spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/PubSubAdmin.java

```diff
@@ -196,8 +200,11 @@ public Subscription createSubscription(String subscriptionName, String topicName
 
 		int finalAckDeadline = this.defaultAckDeadline;
 		if (ackDeadline != null) {
-			Assert.isTrue(ackDeadline >= 0,
-					"The acknowledgement deadline value can't be negative.");
+			Assert.isTrue(ackDeadline >= MIN_ACK_DEADLINE_SECONDS
+							&& ackDeadline <= MAX_ACK_DEADLINE_SECONDS,
+					String.format("The acknowledgement deadline must be between %d and %d seconds.",
```

Comment:
I'm not sure if `String.format` is a great idea here since the compiler can't optimize that.
The regular concatenation with `+` is probably significantly more efficient.

Reply:
Changed.